### PR TITLE
Add preinstall and settings configs

### PR DESCRIPTION
**Pull request by andrew-t-james**
## Description

Add preinstall hook to check node version and a script to run during the preinstall hook. Added settings and extensions for vscode.

## Motivation and Context

This change is needed because we had varying version of node version (potentially out of range for our current supported node version). The extensions and settings will help align vscode users settings so that they match what is recommended for this project.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,7 +86,7 @@ In a terminal window run these commands if you need to update a snapshot.
 
 ```sh
 $ cd vets-who-code-app
-$ npm run test --updateSnapshot
+$ npm run test -- --updateSnapshot
```
Comment (gixxerblade) [inline README.md:89]: `-- --updateSnapshot`?

Comment (andrew-t-james) [reply README.md:89]: Need to pass the command correctly to jest because this is run via npm.

File: scripts/node-engine-check.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      colors.reset,
+      `******* Good to Go with your Node Version:  ${result.node}  *******  `,
+      '\n\n'
+    )
+  } else {
+    console.log(
+      colors.error,
+      ' [ERROR]',
+      colors.reset,
+      `******* Your current Node Version is: ${result.node} and must be >= 14.X.X *******  `,
```
Comment (gixxerblade) [inline scripts/node-engine-check.js:26]: Greater than node 16 is okay according to this error message but L13 says >= 14 || <= 16

Comment (andrew-t-james) [reply scripts/node-engine-check.js:26]: -Good catch will fix

**Review by gixxerblade — APPROVED** [review]

**Review by jeromehardaway — APPROVED** [review]

**Pull request merged**